src/Arcus.WebApi.Security.Authorization/Extensions/FilterCollectionExtensions.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using Arcus.WebApi.Security.Authorization;
+using GuardNet;
+using Microsoft.Extensions.DependencyInjection;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.AspNetCore.Mvc.Filters
+{
+    public static class FilterCollectionExtensions
+    {
+        /// <summary>
+        /// Adds Azure Managed Identity authorization
+        /// </summary>
+        /// <param name="filters">All filters that are being applied to the request pipeline</param>
+        /// <param name="serviceDependencies">Dependencies in the current application</param>
+        public static FilterCollection AddAzureManagedIdentityAuthorization(this FilterCollection filters, IServiceCollection serviceDependencies)
+        {
+            Guard.NotNull(filters, nameof(filters));
+
+            return filters.AddAzureManagedIdentityAuthorization(serviceDependencies, options => { });
+        }
+
+        /// <summary>
+        /// Adds Azure Managed Identity authorization
+        /// </summary>
+        /// <param name="filters">All filters that are being applied to the request pipeline</param>
+        /// <param name="serviceDependencies">Dependencies in the current application</param>
+        /// <param name="options">Configuration options for using Azure Managed Identity authorization</param>
+        public static FilterCollection AddAzureManagedIdentityAuthorization(this FilterCollection filters, IServiceCollection serviceDependencies, Action<AzureManagedIdentityAuthorizationOptions> options)
+        {
+            Guard.NotNull(filters, nameof(filters));
+
+            var azureManagedIdentityAuthorizationOptions = AzureManagedIdentityAuthorizationOptions.Default;
+            options(azureManagedIdentityAuthorizationOptions);
+
+            serviceDependencies.AddTransient(services => azureManagedIdentityAuthorizationOptions);
```

Comment:
Should this be a transient scoped ?  Can possibly be a singleton

Reply:
This can be for sure, I tend to not use singletons except for special reasons but I'm not an expert.

Happy to change if you want.

Reply:
I'd say we can always use the same instance and avoid having new instances be created each time an instance is required.
What is your opinion @stijnmoreels ?

Reply:
I think you should use the `serviceDependencies.Configure` for options. And use `IOptions<>` in the filter.

Reply:
I can do that!